contracts/swap-holder/src/msg.rs

```diff
@@ -102,19 +102,36 @@ pub enum ContractState {
 }
 
 impl ContractState {
-    pub fn complete_and_dequeue(deps: DepsMut) -> StdResult<()> {
+    pub fn complete_and_dequeue(deps: DepsMut, privileged_addr: &str) -> Result<WasmMsg, StdError> {
         CONTRACT_STATE.save(deps.storage, &ContractState::Complete)?;
-        Ok(())
+        dequeue_msg(privileged_addr)
     }
 }
 
+// Local struct used for defining a Dequeue message.
```

Comment:
have you seen the clock helpers file? I think we can just use that one

```rust
use crate::{
    error::ContractError,
    msg::ExecuteMsg::{Dequeue, Enqueue},
};
use cosmwasm_std::{to_json_binary, Addr, StdResult, WasmMsg};
use neutron_sdk::NeutronError;

pub fn enqueue_msg(addr: &str) -> StdResult<WasmMsg> {
    Ok(WasmMsg::Execute {
        contract_addr: addr.to_string(),
        msg: to_json_binary(&Enqueue {})?,
        funds: vec![],
    })
}

pub fn dequeue_msg(addr: &str) -> StdResult<WasmMsg> {
    Ok(WasmMsg::Execute {
        contract_addr: addr.to_string(),
        msg: to_json_binary(&Dequeue {})?,
        funds: vec![],
    })
}

pub fn verify_clock(caller: &Addr, clock_addr: &Addr) -> Result<(), NeutronError> {
    if caller != clock_addr {
        return Err(ContractError::NotClock.into());
    }

    Ok(())
}
```

Reply:
Yes, but that would mean adding back the dependency to the clock contract ..

Reply:
Shall we move the clock helpers to covenant-utils then? Not convinced that's the move but curious what you think

Reply:
I think it makes some sense. We could move that into a `clock.rs` module (in covenant-utils). That would already break the direct dependencies from contracts to the clock. Now, the covenant-utils is a bit of mixed bag of many different things, and should probably be cleaned up at some point, but this can come later. wdyt ?
I'll create a separate technical task to move the helpers to covenant-utils.

Reply:
https://github.com/timewave-computer/covenants/issues/268